File: dart/src/n02_test_data_generator/main.dart
import 'dart:io';

import 'data_generator.dart';
import 'data_set_saver_csv.dart';
import 'data_set_saver_json.dart';
import 'data_set_saver_xml.dart';

void main() async {
  final DataGenerator dataGenerator = DataGenerator();

  while_input:
  while (true) {
    stdout.write('Output as csv, json or xml (c, j, x): ');
    final String? input = stdin.readLineSync();
    switch (input) {
      case 'csv':
      case 'c':
        dataGenerator.saver = DataSetSaverCSV();
        break while_input;
      case 'json':
      case 'j':
        dataGenerator.saver = DataSetSaverJSON();
        break while_input;
      case 'xml':
      case 'x':
        dataGenerator.saver = DataSetSaverXML();
        break while_input;
      default:
        print('Invalid input');
        break;
    }
  }

  stdout.write('Output filename: ');
  final String input = stdin.readLineSync() ?? '';
  final File outputFile = File(input);

  await (dataGenerator..createData(0, 10)).saveData(outputFile);
}

File: java/src/n02_test_data_generator/Main.java
package n02_test_data_generator;

import java.io.File;

public class Main {
    public static void main(String[] args) {
        DataGenerator dataGenerator = new DataGenerator();

        while_input:
        while (true) {
            System.out.print("Output as csv, json or xml (c, j, x): ");
            String input = System.console().readLine();
            switch (input.toLowerCase()) {
                case "csv", "c":
                    dataGenerator.setDataSetSaver(new DataSetSaverCSV());
                    break while_input;
                case "json", "j":
                    dataGenerator.setDataSetSaver(new DataSetSaverJSON());
                    break while_input;
                case "xml", "x":
                    dataGenerator.setDataSetSaver(new DataSetSaverXML());
                    break while_input;
                default:
                    System.out.println("Invalid input");
                    break;
            }
        }

        System.out.print("Output filename: ");
        String input = System.console().readLine();
        File outputFile = new File(input);

        dataGenerator.createData(0, 10);
        dataGenerator.saveData(outputFile);
    }
}
